### PR TITLE
Feature/node 43 API s for organisation management

### LINKED ISSUE #43: [node-backend] - APIs for organisation management
**APIs needed:**
- [x] Create organisation - POST /organisations

- [x] Update organisation - PUT /organisations/:id

- [x] List organisations - GET /organisations

- [x]  Show organisation - GET /organisations/:id

**Authentication and Authorization:**

Only SuperAdmin is allowed with a valid JWT token

### PR DESCRIPTION
**Pull request by jitendrasbunde**
Closes #43

### CODE REVIEW

File: node-backend/app/controllers/orgnizationController.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
+const db = require("../models/sequelize");
+const Organizations = db.organizations;
+const yup = require("yup");
```
Comment (sunil4sonawane) [inline node-backend/app/controllers/orgnizationController.js:3]: yup should be included at top then give new line and then add your code modules. Third party modules and our code modules must be separated and  grouped

File: node-backend/app/controllers/orgnizationController.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+const schema = yup.object().shape({
+  name: yup.string().required(),
+  contact_email: yup.string().email(),
```
Comment (sunil4sonawane) [inline node-backend/app/controllers/orgnizationController.js:7]: how it satisfies valid email validation?

File: node-backend/app/controllers/orgnizationController.js [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+  domain_name: yup.string().required(),
```
Comment (sunil4sonawane) [inline node-backend/app/controllers/orgnizationController.js:8]: how it satisfies valid domain validation?

File: node-backend/app/controllers/orgnizationController.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        .then((info) => {
+          res.send({
+            status: 201,
+            data: info,
+          });
+        })
+        .catch((err /*eslint-disable-line no-unused-vars*/) => {
+          res.send({
+            status: 500,
+            message: "Some error occurred while creating the Organization.",
```
Comment (sunil4sonawane) [inline node-backend/app/controllers/orgnizationController.js:43]: Please follow API doc from WIki for correct response format and error messages

File: node-backend/app/controllers/orgnizationController.js [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+          });
+        });
+    } else {
+      res.send({
+        status: 400,
+        message: "Contents not be correct format",
```
Comment (sunil4sonawane) [inline node-backend/app/controllers/orgnizationController.js:49]: It should provide which parameters are in incorrect format

File: node-backend/app/controllers/orgnizationController.js [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+exports.findAll = /*eslint-disable-line node/exports-style*/ (req, res) => {
+  Organizations.findAll()
+    .then((info) => {
+      res.json({
+        status: 200,
+        data: info,
+      });
+    })
+    .catch((err /*eslint-disable-line no-unused-vars*/) => {
```
Comment (sunil4sonawane) [inline node-backend/app/controllers/orgnizationController.js:63]: why ypu don't add eslint statement out side of brackets?

File: node-backend/app/models/organizations.model.js [hunk 7]
```diff
@@ -1,47 +1,55 @@
...
+        type: Sequelize.STRING,
+        length: 45,
+        notNull: true,
+      },
+      subscription_status: {
+        type: Sequelize.INTEGER,
+        notNull: true,
+      },
+      subscription_valid_upto: {
+        type: "TIMESTAMP",
```
Comment (sunil4sonawane) [inline node-backend/app/models/organizations.model.js:31]: I asked to change TIMESTAMP to big int every where

File: node-backend/app/routes/organisation.routes.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
+const express = require("express");
+const bodyParser = require("body-parser");
+const organisationController = require("../controllers/orgnizationController");
+const router = express.Router();
```
Comment (sunil4sonawane) [inline node-backend/app/routes/organisation.routes.js:4]: should be orgRouter

File: node-backend/app/test/orgnization.test.js [hunk 9]
```diff
@@ -0,0 +1,157 @@
+const supertest = require("supertest"); //eslint-disable-line node/no-unpublished-require
+const should = require("should" /*eslint-disable-line node/no-unpublished-require*/); //eslint-disable-line no-unused-vars
+
+// This agent refers to PORT where program is runninng.
+
+const server = supertest.agent("http://localhost:3120");
```
Comment (sunil4sonawane) [inline node-backend/app/test/orgnization.test.js:6]: URL hard coded?

File: node-backend/app/controllers/orgnizationController.js (hunk 3 above)
Comment (jitendrasbunde) [reply node-backend/app/controllers/orgnizationController.js:8]: can you examples of domain name ?

File: node-backend/app/controllers/orgnizationController.js [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+      "Invalid domain name"
+    )
+    .required("Domain name is required"),
+  subscription_status: yup.number().required("Subscription status is required"),
+  subscription_valid_upto: yup
+    .number()
+    .required("Subscription valid upto is required"),
+  hi5_limit: yup.number().required("Hi5 limit is required"),
+  hi5_quota_renewal_frequency: yup.string(
+    "Hi5 quota renewal frequency is empty"
```
Comment (sunil4sonawane) [inline node-backend/app/controllers/orgnizationController.js:22]: "Hi5 quota renewal frequency is required"

File: node-backend/app/controllers/orgnizationController.js [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+      Organizations.findByPk(id)
+        .then((info) => {
+          if (info !== null) {
+            res.status(200).send({
+              data: info,
+            });
+          } else {
+            res.status(404).send({
+              error: {
+                message: "Not Found",
```
Comment (sunil4sonawane) [inline node-backend/app/controllers/orgnizationController.js:101]: Organisation with specified id is not found

File: node-backend/app/controllers/orgnizationController.js [hunk 12]
```diff
@@ -0,0 +1,185 @@
...
+            error: {
+              message: "internal server error",
+            },
+          });
+        });
+    })
+    .catch((err /*eslint-disable-line no-unused-vars*/) => {
+      res.status(400).send({
+        error: {
+          message: ["Invalid Id"],
```
Comment (sunil4sonawane) [inline node-backend/app/controllers/orgnizationController.js:117]: Invalid value for parameter id

File: node-backend/app/controllers/orgnizationController.js [hunk 13]
```diff
@@ -0,0 +1,185 @@
...
+            })
+              .then((num) => {
+                if (num == 1) {
+                  res.status(200).send({
+                    message: "Organization is updated successfully.",
+                  });
+                } else {
+                  res.send({
+                    status: 404,
+                    message: `Cannot update Organization. Maybe Organization was not found`,
```
Comment (sunil4sonawane) [inline node-backend/app/controllers/orgnizationController.js:164]: Organisation with specified id not found

File: node-backend/app/controllers/orgnizationController.js [hunk 14]
```diff
@@ -0,0 +1,186 @@
...
+            error: {
+              message: "internal server error",
+            },
+          });
+        });
+    })
+    .catch((err /*eslint-disable-line no-unused-vars*/) => {
+      res.status(400).send({
+        error: {
+          message: ["Invalid value for parameter id"],
```
Comment (sunil4sonawane) [inline node-backend/app/controllers/orgnizationController.js:117]: why this is array?

**Pull request merged**